pkg/util/leaktest/leaktest.go

```diff
@@ -78,40 +79,68 @@ func interestingGoroutines() map[int64]string {
 	return gs
 }
 
+// Set once a test leaks goroutines so that further tests don't attempt to
+// detect leaks any more. Once a tests leaks, it has soiled the process beyond
+// repair: even though other tests would take a snapshot of goroutines at the
+// beginning that would include the previously-leaked goroutines, those leaked
+// goroutines can spin up other goroutines at random times and these would be
+// mis-attributed as leaked by the currently-running test.
+var leakDetectorDisabled bool
+
 // AfterTest snapshots the currently-running goroutines and returns a
 // function to be run at the end of tests to see whether any
 // goroutines leaked.
 func AfterTest(t testing.TB) func() {
```

Comment:
@RaduBerinde I think we'd just return here if `leakDetectorDisabled`. Oversight, sorry about that!